pkgs/applications/science/electronics/horizon-eda/default.nix

```diff
@@ -29,47 +32,70 @@ stdenv.mkDerivation rec {
     owner = "horizon-eda";
     repo = "horizon";
     rev = "v${version}";
-    sha256 = "sha256-UcjbDJR6shyETpanNkRoH8LF8r6gFjsyNHVSCMHKqS8=";
+    hash = "sha256-UcjbDJR6shyETpanNkRoH8LF8r6gFjsyNHVSCMHKqS8=";
   };
 
   buildInputs = [
     cppzmq
     curl
-    libepoxy
     glm
     gtkmm3
     libarchive
+    libepoxy
     libgit2
     librsvg
-    libspnav
     libuuid
     opencascade-occt
     podofo
-    python3
     sqlite
+  ] ++ lib.optionals (!buildPythonPackage) [
+    libspnav
+  ] ++ lib.optionals (buildPythonPackage) [
+    mesa
+    mesa.osmesa
+    python3Packages.python
   ];
 
   nativeBuildInputs = [
-    boost.dev
     pkg-config
+  ] ++ lib.optionals (!buildPythonPackage) [
+    python3
+    boost.dev
     wrapGAppsHook
   ];
 
-  CASROOT = opencascade-occt;
+  propagatedBuildInputs = lib.optionals (buildPythonPackage) [
+    python3Packages.pycairo
+  ];
 
-  installFlags = [
+  env.CASROOT = opencascade-occt;
+
+  buildFlags = lib.optionals (buildPythonPackage) [
+    "pymodule"
+  ];
+
+  installFlags = lib.optionals (!buildPythonPackage) [
     "INSTALL=${coreutils}/bin/install"
     "DESTDIR=$(out)"
     "PREFIX="
   ];
 
+  installPhase = lib.optionalString (buildPythonPackage) ''
+    runHook preInstall
+
+    mkdir -p $out/${python3Packages.python.sitePackages}
+    cp build/horizon.so $out/${python3Packages.python.sitePackages}
+
+    runHook postInstall
+  '';
```

Comment:
I'm not sure about this expression, it looks that `installPhase` is empty for the non-Python package.

Reply:
> I'm not sure about this expression, it looks that `installPhase` is empty for the non-Python package.

If the `installPhase` is empty, it should pick the default one provided by the builder, or am I wrong? At least I'm able to build the binary and the produced `result` has everything I expect to be present.

Currently, horizon-eda has no `install` target for the python module. We can't get around ugly solutions, I guess?

> I'd suggest putting most of the logic in `// lib.optionalAttrs buildPythonPackage { ... }` after meta, or we just go back to [2a97676](https://github.com/NixOS/nixpkgs/commit/2a97676bc1f456b0409344383b463d003c425eb9). What do you think?

Idk. I'm a newbie and it's hard for me to tell which one is the better solution. I would do every change you ask for if this helps to bring this one upstream.

Reply:
`lib.optionalString false "..."` would evaluate to the empty string instead of `null`, so I don't expect this would work.